exercises/custom-set/test/Tests.hs

```diff
@@ -1,92 +1,204 @@
+
+        it "adding an existing element does not change the set" $
+          insert 3 (fromList [1, 2, 3]) `shouldBe` fromList [1, 2, 3]
+
+      describe "intersection" $ do
+
+        it "intersection of two empty sets is an empty set" $
+          fromList ([] :: [Integer]) `intersection` fromList [] `shouldBe` fromList []
+
+        it "intersection of an empty set and non-empty set is an empty set" $
+          fromList [] `intersection` fromList [3, 2, 5] `shouldBe` fromList []
+
+        it "intersection of a non-empty set and an empty set is an empty set" $
+          fromList [1, 2, 3, 4] `intersection` fromList [] `shouldBe` fromList []
+
+        it "intersection of two sets with no shared elements is an empty set" $
+          fromList [1, 2, 3] `intersection` fromList [4, 5, 6] `shouldBe` fromList []
+
+        it "intersection of two sets with shared elements is a set of the shared elements" $
+          fromList [1, 2, 3, 4] `intersection` fromList [3, 2, 5] `shouldBe` fromList [2, 3]
+
+      describe "difference" $ do
+
+        it "difference of two empty sets is an empty set" $
+          fromList ([] :: [Integer]) `difference` fromList [] `shouldBe` fromList []
+
+        it "difference of empty set and non-empty set is an empty set" $
+          fromList [] `difference` fromList [3, 2, 5] `shouldBe` fromList []
+
+        it "difference of a non-empty set and an empty set is the non-empty set" $
+          fromList [1, 2, 3, 4] `difference` fromList [] `shouldBe` fromList [1, 2, 3, 4]
+
+        it "difference of two non-empty sets is a set of elements that are only in the first set" $
+          fromList [3, 2, 1] `difference` fromList [2, 4] `shouldBe` fromList [1, 3]
+
+      describe "union" $ do
+
+        it "union of empty sets is an empty set" $
+          fromList ([] :: [Integer]) `union` fromList [] `shouldBe` fromList []
+
+        it "union of an empty set and non-empty set is the non-empty set" $
+          fromList [] `union` fromList [2] `shouldBe` fromList [2]
+
+        it "union of a non-empty set and empty set is the non-empty set" $
+          fromList [1, 3] `union` fromList [] `shouldBe` fromList [1, 3]
+
+        it "union of non-empty sets contains all unique elements" $
+          fromList [1, 3] `union` fromList [2, 3] `shouldBe` fromList [3, 2, 1]
+
+    describe "track-specific tests" $ do
+
+      -- Track-specific test cases.
+
+      describe "delete" $ do
+
+        it "delete existing element" $
+          delete 2 (fromList [1, 2, 3]) `shouldBe` fromList [1, 3]
+
+      describe "size" $ do
+
+        it "size of an empty set is zero" $
+          size (fromList ([] :: [Integer])) `shouldBe` 0
+
+        it "size is the number of elements" $
+          size (fromList [1, 2, 3, 2]) `shouldBe` 3
+
+        it "size doesn't count repetitions in `fromList`" $
+          size (fromList [1, 2, 3, 2]) `shouldBe` 3
+
+        it "size doesn't count repetitions from `insert`" $
```

Comment:
Thank you very much for the review! The `size` tests where really wrong! :smile:

I think it is fixed now. :+1: :question: